dhall/src/Dhall/Core.hs

```diff
@@ -2283,12 +2298,23 @@ multiLet x0 mA0 a0 = \case
 > wrapInLets as b1 == Let x mA a b0
 -}
 wrapInLets :: Foldable f => f (Binding s a) -> Expr s a -> Expr s a
-wrapInLets as b = foldr (\(Binding x mA a) e -> Let x mA a e) b as
+wrapInLets bs e = foldr Let e bs
 
 data MultiLet s a = MultiLet (NonEmpty (Binding s a)) (Expr s a)
 
 data Binding s a = Binding
-    { variable   :: Text
-    , annotation :: Maybe (Expr s a)
-    , value      :: Expr s a
-    }
+    { bindingSrc0 :: Maybe s
+    , variable    :: Text
+    , bindingSrc1 :: Maybe s
+    , annotation  :: Maybe (Maybe s, Expr s a)
+    , bindingSrc2 :: Maybe s
+    , value       :: Expr s a
+    } deriving (Data, Eq, Foldable, Functor, Generic, NFData, Ord, Show, Traversable)
```

Comment:
There seems to be an invariant that either all of the `Maybe s` are `Just` or all of them are `Nothing`. I guess it doesn't matter though.

Reply:
Yeah, I couldn't figure out a better way to enforce that consistency at the type-level